src/ex5/server/services/itemManager.js

```diff
@@ -0,0 +1,168 @@
+const pokemonClinet = require("../clients/pokemonClient.js");
+const { items } = require("../db/models");
+
+class ItemManager {
+  constructor() {
+    this.itemsArray = [];
+    this.taskToDbPath = "../server/db/taskToDb.json";
+  }
+
+  async getAll() {
+    try{
+      this.itemsArray = await items.findAll({raw : true});
+      return this.itemsArray;
+    }catch(err){
+      throw new Error(err)
+    }
+  }
+
+  async deleteAll() {
+    try{
+    this.itemsArray = [];
+    await items.destroy({
+      where :{},
+      truncate : true,
+    });
+  }
+    catch(err){
+    console.log('Error While delete an Item');
+    throw new Error(err) 
+    
+  }
+}
+  async checkByPokemonName(pokemon,itemToRender) {
+   try{
+    const pokemonIsExist = await this.exsitInDb(pokemon.id);
+    if (!pokemonIsExist){
+      const task = this.pokemonInit(
+        true,
+        pokemon.name,
+        pokemon.sprites.front_default,
+        pokemon.id
+      );
+    this.itemsArray.push(task);
+    await items.bulkCreate([task]);
+    return [task];
+   }
+   return itemToRender;
+  }
+  catch(err){
+  throw new Error(err);
+  }
+  }
+
+  async getPokemonById(filteredArr,itemToRender) {
+    try {
+      const pokemons = await pokemonClinet.fetchPokemon(filteredArr);
+
+      pokemons.forEach((pokemon) => {
+        const task = this.pokemonInit(
+          true,
+          pokemon.name,
+          pokemon.sprites.front_default,
+          pokemon.id
+        );
+        this.itemsArray.push(task);
+        itemToRender.push(task);
+      });
+      await items.bulkCreate(itemToRender);
+    } catch (e) {
+      let pokemonId = "";
+      filteredArr.forEach((task) => {
+        pokemonId += task + " ";
+      });
+      const task = this.pokemonInit(
+        false,
+        `pokemon with id: ${pokemonId} was not found`
+      );
+      this.itemsArray.push(task);
+      itemToRender.push(task);
+      await items.bulkCreate(itemToRender);
+    }
+    return itemToRender;
+  }
+  // generate UUID number
+  uuidGeneration() {
```

Comment:
1. better use open source library that already do it if possible
2. not the best service to have this code in, better move it to util file